gprofiler/log.py

```diff
@@ -125,6 +127,12 @@ def emit(self, record: LogRecord) -> None:
             return
 
         self._logs.append(self._make_dict_record(record))
+        # truncate logs to last N entries
+        if len(self._logs) > self.MAX_BUFFERED_RECORDS:
+            self._logger.warn(
+                f"Truncating log buffer as the maximum number of records ({self.MAX_BUFFERED_RECORDS}) has been reached"
+            )
```

Comment:
```suggestion
            self._logger.warning(
                f"Truncating log buffer as the maximum number of records ({self.MAX_BUFFERED_RECORDS}) has been reached"
            )
```
`warn` is deprecated